lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -127,10 +133,25 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult, TPartitionType>
         private int _numberOfRetries;
 
         /// <summary>
+        /// Minimum timeout in milliseconds for TaskWaitingForClose
+        /// </summary>
+        private readonly int _minTaskWaitingForCloseTimeout;
+
+        /// <summary>
         /// Manages lifecycle events for driver, like JobCancelled event.
         /// </summary>
         private readonly List<IDisposable> _disposableResources = new List<IDisposable>();
 
+        private Timer _timeputMonitorTimer;
+
+        private readonly object _disposeLock = new object();
+
+        /// <summary>
+        /// Record tasks that are closed after timeout. 
+        /// The CompletedTask and failedEvaluator events from those tasks should be ignored to avoid double counted. 
+        /// </summary>
+        private readonly IList<string> _taskForceClosed = new List<string>();
```

Comment:
We're not actually closing the tasks, we're killing underlying evaluators. Would it make sense to keep a list of killed evaluators, so that any event that comes from any of these evaluators is ignored? I suspect this will be handy later when we're implementing timeouts for other events, since in all cases our reaction will be to kill evaluator, not the task